net/sslh/src/opnsense/mvc/app/models/OPNsense/Sslh/Settings.xml

```diff
@@ -0,0 +1,66 @@
+<model>
+    <mount>//OPNsense/sslh</mount>
+    <version>0.0.1</version>
+    <items>
+        <enabled type="BooleanField">
+            <required>Y</required>
+        </enabled>
+        <listen_addresses type="CSVListField">
+            <required>Y</required>
+            <default>localhost:443</default>
+            <validationmessage>Please enter at least one hostname/IP:port combination.</validationmessage>
```

Comment:
validation message here but no special validation of individual values?

Reply:
localhots:443 is probably already bound to web GUI btw

Reply:
For this, I considered the options that I'm aware of and can understand, but I'm having a challenge understanding the best approach. I could use something like a regular expression, but the fact that this supports a hostname and IP address will make this complicated to do in single mask.

I thought about a custom FieldType with a new validator using the CallbackValidator, the challenges would be similar, but I could at least split each CSV and explode on the colon (only reliable for IPv4), and evaluate each half of each entry separately. I feel that's still re-inventing the wheel when there exists HostValidator, and PortField already has a validator (though hard coded in the FieldType itself).

I think an option might be to re-factor the field into an ArrayField, and specify the hostname/IP, and port separately and validate each according to each field type. However, I'd prefer not to do that as it would require building the interface(s) to support the user input. I wasn't planning on going down that route until the next version of the plugin.

I'm not strong in the validators area, and I left this alone due to the complexity of specifying a single mask to handle the entire thing. Let me know how this could be approached in a reasonable manner.